src/link.cpp

```diff
@@ -898,10 +898,44 @@ namespace mamba
             return std::make_tuple(validate::sha256sum(dst), rel_dst);
         }
 
-        if (path_data.path_type == PathType::HARDLINK)
+        if (path_data.no_link || (path_data.path_type == PathType::HARDLINK))
         {
-            LOG_INFO << "hard linked " << src << " --> " << dst;
-            fs::create_hard_link(src, dst);
+            bool copy = path_data.no_link || m_context->always_copy;
+            bool softlink = m_context->always_softlink;
+
+            if (!copy && !softlink)
+            {
+                std::error_code ec;
+                fs::create_hard_link(src, dst, ec);
+
+                if (ec)
+                {
+                    softlink = m_context->allow_softlinks;
+                    copy = !softlink;
+                }
+                else
+                {
+                    LOG_INFO << "hard linked " << src << " --> " << dst;
```

Comment:
return here? otherwise it can _also_ softlink?

Reply:
ok i think i get it now, looks fine!

Reply:
cool thanks for the review!